xapian-core/cluster/cluster.cc

```diff
@@ -81,11 +102,19 @@ TermListGroup::add_document(const Document &document)
 
     TermIterator titer(document.termlist_begin());
 
+    bool allow_stopword_removal = stopper.get() ? true : false;
     for (; titer != document.termlist_end(); ++titer) {
+	const string &term = *titer;
+
+	// Remove stopwords by using the Xapian::Stopper object
+	// that is set
+	if (allow_stopword_removal && (*stopper)(term))
```

Comment:
I'd scrap the flag and just test `if (stopper.get() && (*stopper)(term))`.  It's clearer to the reader what's happening, and that the potential `NULL` dereference is guarded against, and it should be efficient - `opt_intrusive_ptr`'s `get()` method is inlined and just returns the `px` while `operator*` is inlined and returns `*px`, so this boils down to `if (px && (*px)(term))`.
 